rlbot_gui/gui.py

```diff
@@ -464,15 +464,39 @@ def download_bot_pack():
     botpack_location = content_folder / BOTPACK_FOLDER
 
     # The bot pack in now hosted at https://github.com/RLBot/RLBotPack
-    BotpackDownloader().download(BOTPACK_REPO_OWNER, BOTPACK_REPO_NAME, BOTPACK_REPO_BRANCH, botpack_location)
+    success = BotpackDownloader().download(BOTPACK_REPO_OWNER, BOTPACK_REPO_NAME, BOTPACK_REPO_BRANCH, botpack_location)
 
-    # Configure the folder settings.
-    bot_folder_settings['folders'][str(botpack_location)] = {'visible': True}
+    if success:
+        # Configure the folder settings.
+        bot_folder_settings['folders'][str(botpack_location)] = {'visible': True}
 
-    settings = load_settings()
-    settings.setValue(BOT_FOLDER_SETTINGS_KEY, bot_folder_settings)
-    settings.setValue(COMMIT_ID_KEY, get_last_botpack_commit_id())
-    settings.sync()
+        settings = load_settings()
+        settings.setValue(BOT_FOLDER_SETTINGS_KEY, bot_folder_settings)
+        settings.setValue(COMMIT_ID_KEY, get_last_botpack_commit_id())
+        settings.sync()
+        scan_for_bots()
+
+
+@eel.expose
+def update_bot_pack():
+    content_folder = get_content_folder()
+    botpack_location = content_folder / BOTPACK_FOLDER
+
+    # The bot pack in now hosted at https://github.com/RLBot/RLBotPack
+    success = BotpackUpdater().update(BOTPACK_REPO_OWNER, BOTPACK_REPO_NAME, BOTPACK_REPO_BRANCH, botpack_location)
+
+    if success == "download":
+        success = BotpackDownloader().download(BOTPACK_REPO_OWNER, BOTPACK_REPO_NAME, BOTPACK_REPO_BRANCH, botpack_location)
+
+    if success:
```

Comment:
I don't like that the update() function sometimes returns "download" and sometimes returns a boolean. Please refactor somehow.